regulation-worker/internal/model/model.go

```diff
@@ -6,19 +6,23 @@ import (
 )
 
 var (
-	ErrDestTypeNotFound = errors.New("destination type not found for the destination ID")
-	ErrDestDetail       = errors.New("error while getting destination details")
-	ErrNoRunnableJob    = errors.New("no runnable job found")
+	ErrDestTypeNotFound   = errors.New("destination type not found for the destination ID")
+	ErrDestDetail         = errors.New("error while getting destination details")
+	ErrNoRunnableJob      = errors.New("no runnable job found")
+	ErrDestNotImplemented = errors.New("job deletion not implemented for the destination")
 )
 
 type JobStatus string
 
 const (
-	JobStatusUndefined JobStatus = ""
-	JobStatusPending   JobStatus = "pending"
-	JobStatusRunning   JobStatus = "running"
-	JobStatusComplete  JobStatus = "complete"
-	JobStatusFailed    JobStatus = "failed"
+	JobStatusUndefined         JobStatus = ""
+	JobStatusPending           JobStatus = "pending"
+	JobStatusRunning           JobStatus = "running"
+	JobStatusComplete          JobStatus = "complete"
+	JobStatusFailed            JobStatus = "failed"
+	JobStatusNotSupported      JobStatus = "not supported"
+	JobStatusInvalidFormat     JobStatus = "invalid format"
+	JobStatusInvalidCredential JobStatus = "invalid credential"
```

Comment:
What if we use `JobStatusFailed` in place of invalid format/ credential, but also include an error code/message that explains the reason.

Reply:
I was thinking of updating the current status shown in regulation-manager DB with all of these to have more transparency. 
I am planning to log all the errors. 
Given above two points, I am not sure, to whom passing the error would be helpful.